### PR TITLE
improve eval code

### PR DESCRIPTION
**Pull request by jordan-definitive**
* split up "request/response" and "validation" flows.

### CODE REVIEW

**Review by cwelton — COMMENTED** [review]
Fails lint checks.  Run `invoke check` to check a variety of different issues.

File: packages/openassistants/openassistants/eval/checks/function_selection.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from openassistants.eval.v2 import (
```
Comment (cwelton) [inline packages/openassistants/openassistants/eval/checks/function_selection.py:1]: I'm having difficulty identifying where `v2` comes from here, I don't see a `v2` in `packages/openassistants/openassistants/eval/`

When I run from poetry shell:

```
from openassistants.eval.v2 import BaseResponseChecker
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'openassistants.eval.v2'
```

This looks like a bug.

File: packages/openassistants/openassistants/eval/eval_interaction.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
+import abc
```
Comment (cwelton) [inline packages/openassistants/openassistants/eval/eval_interaction.py:1]: Is there a reason we switched from `openassistants.eval.interaction` to `openassistants.eval.eval_interaction`?  I thought the previous filename was reasonable, and don't know what this refactor does other than make the diff harder to reason over.

**Review by cwelton — CHANGES_REQUESTED** [review]
I believe the `v2` package import is broken.

Please add tests and fix lint issues.

File: packages/openassistants/openassistants/eval/eval_interaction.py (hunk 2 above)
Comment (jordan-definitive) [reply packages/openassistants/openassistants/eval/eval_interaction.py]: fixed. ugh pre commit hooks / static analysis are not catching these.

File: packages/openassistants/openassistants/eval/checks/function_selection.py (hunk 1 above)
Comment (jordan-definitive) [reply packages/openassistants/openassistants/eval/checks/function_selection.py]: same as below

**Comment by jordan-definitive** [comment]
fixed lints / bugs

File: examples/fast-api-server/tests/test_piedpiper.py [hunk 3]
```diff
@@ -24,8 +28,34 @@
...
 async def test_interaction():
-    assistant = Assistant(libraries=["piedpiper"])
-    await run_interaction(core_interaction, assistant)
+    response = await core_interaction.run(hooli_assistant, [])
+    report = await response.run_checks(core_checks)
+    print(report.pretty_repr())
+
+
+@pytest.mark.asyncio
+async def test_entity_resolution():
```
Comment (jordan-definitive) [inline examples/fast-api-server/tests/test_piedpiper.py:58]: this test "runs" the entity resolution

File: packages/openassistants/openassistants/eval/interaction.py [hunk 4]
```diff
@@ -19,322 +14,232 @@
...
-            children=[
-                InteractionReport.cascade_failure(c, "cascading failure from parent")
-                for c in interaction.children
-            ],
+    async def run_function_selection(
+        self,
+        assistant: Assistant,
+        history_with_user_message: List[OpasMessage],
+    ) -> OpasAssistantMessage:
+        response_messages = await last_value(
```
Comment (cwelton) [inline packages/openassistants/openassistants/eval/interaction.py:56]: would be nice to add a type annotation here and on the other places in this file you set response_messages.

**Review by cwelton — APPROVED** [review]
- You do introduce 3 new mypy typing issues, but they are minor
- Still didn't add tests, but given that I can't spot any problems I can let that slide because our baseline (no tests at all) is terrible, this doesn't make things worse, and looks to have some good improvements.

**Review by cwelton — APPROVED** [review]

**Review by colemccracken — APPROVED** [review]

File: packages/openassistants/tasks.py [hunk 5]
```diff
@@ -8,8 +8,8 @@
 @task
 def refresh_poetry(ctx):
     with ctx.cd(PROJECT_DIR):
-        ctx.run('rm -r .venv poetry.lock')
-        ctx.run('poetry install --with test', pty=True)
+        ctx.run("rm -r .venv poetry.lock")
```
Comment (colemccracken) [inline packages/openassistants/tasks.py:11]: Is this a formatter issue?

Comment (cwelton) [reply packages/openassistants/tasks.py:11]: Yes, I believe ruff does this.

**Pull request merged**